### PR TITLE
Feature/transfer class metadata epmdj 10603

### PR DESCRIPTION
**Pull request by iryabov**

### CODE REVIEW

File: plugin-api-agent/src/commonMain/kotlin/com/epam/drill/plugin/api/processing/AgentPart.kt [hunk 1]
```diff
@@ -38,9 +38,6 @@ abstract class AgentPart<A>(
     //TODO remove from API - this is only used in agent
     open fun load(on: Boolean) {
         initPlugin()
-        if (on) {
```
Comment (RomanDavlyatshin) [inline plugin-api-agent/src/commonMain/kotlin/com/epam/drill/plugin/api/processing/AgentPart.kt:41]: reason for removing `on()` call

File: plugin-api-admin/src/commonMain/kotlin/com/epam/drill/plugin/api/AdminData.kt [hunk 2]
```diff
@@ -16,10 +16,4 @@
 package com.epam.drill.plugin.api
 
 interface AdminData {
```
[the remark this answers is not in the document]
Comment (iryabov) [reply plugin-api-admin/src/commonMain/kotlin/com/epam/drill/plugin/api/AdminData.kt:18]: I think it's possible, it needs some refactoring

File: plugin-api-agent/src/commonMain/kotlin/com/epam/drill/plugin/api/processing/AgentPlugin.kt [hunk 3]
```diff
@@ -49,3 +49,34 @@ interface Instrumenter {
...
+     * Scan target classes of the application under test
+     * @param consumer the function for processing chunks of scanned classes
+     */
+    fun scanClasses(consumer: (Set<EntitySource>) -> Unit)
+}
+
+/**
+ * Class containing information about entity/class bytes
+ */
+interface ByteSource {
```
[the remark this answers is not in the document]
Comment (iryabov) [reply plugin-api-agent/src/commonMain/kotlin/com/epam/drill/plugin/api/processing/AgentPlugin.kt:67]: I have not found a use for ByteSource outside of EntitySource in those modules that we changed. I can try to remove the extra abstraction ByteSource

File: plugin-api-agent/src/commonMain/kotlin/com/epam/drill/plugin/api/processing/AgentPart.kt (hunk 1 above)
Comment (iryabov) [reply plugin-api-agent/src/commonMain/kotlin/com/epam/drill/plugin/api/processing/AgentPart.kt:41]: The load method is used only with the argument "on" equal to true. But that was not the reason for the removal. In fact, the real reason is no longer relevant, and it is possible to return this block of code, or remove the "load" method, as indicated in the comment above.

**Pull request merged**